File: llm_generated_code/ChatGPT/Python/Level1/Q38.py
def max_points(n, lamps):
    lamps.sort(key=lambda x: -x[1])
    points = 0
    on_count = 0
    for a, b in lamps:
        if on_count < a:
            points += b
            on_count += 1
    return points

t = int(input())
results = []
for _ in range(t):
    n = int(input())
    lamps = [tuple(map(int, input().split())) for _ in range(n)]
    results.append(max_points(n, lamps))

for result in results:
    print(result)

File: llm_generated_code/ChatGPT/CPP/Level1/Q38.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

struct Lamp {
    int a, b;
};

bool compareByPoints(const Lamp &l1, const Lamp &l2) {
    return l1.b > l2.b;
}

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;
        
        vector<Lamp> lamps(n);
        for (int i = 0; i < n; i++) {
            cin >> lamps[i].a >> lamps[i].b;
        }
        
        sort(lamps.begin(), lamps.end(), compareByPoints);
        
        int points = 0;
        int turnedOnCount = 0;

        for (int i = 0; i < n; i++) {
            if (lamps[i].a > turnedOnCount) {
                points += lamps[i].b;
                turnedOnCount++;
            }
        }
        
        cout << points << endl;
    }

    return 0;
}
